server/conn.go

```diff
@@ -1546,6 +1553,11 @@ func (cc *clientConn) writeError(ctx context.Context, e error) error {
 // serverStatus, a flag bit represents server information
 // in the packet.
 func (cc *clientConn) writeEOF(serverStatus uint16) error {
+	if cc.capability&mysql.ClientDeprecateEOF > 0 {
+		logutil.Logger(context.Background()).Error(
+			"Attempt to write EOF to client that has the ClientDeprecateEOF flag set", zap.Stack("stack"))
```

Comment:
Why adding this error message?

Reply:
That is not strictly needed, but when `ClientDeprecateEOF` is active this code shouldn't be reached, if it is there is likely a bug, so I added this to make that visible. We could remove this.

Reply:
@bb7133 what do you think? keep this? remove this? Could you suggest other reviewers as well?

Reply:
I think keeping is harmless.